pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -49,7 +49,9 @@ def full(size, fill_value, *, out=None, dtype=None, layout=None, device=None, re
       raise NotImplementedError
     else:
       dtype = torch.get_default_dtype()
-  return zeros(*size, dtype=dtype) + fill_value
+  out = zeros(*size, dtype=dtype) + fill_value
+  out.fill_(fill_value)
```

Comment:
This should not be done. This assumes a kind of strange behavior of `fill_`. This also should be done automatically already in case `fill_value` is constant.

Reply:
What else do you suggest? This is not done automatically yet apparently.

Reply:
But then we should fix this. Why is this not done automatically? This is wrong when this is a constant.

Reply:
In `Module.make_output_tensor_from_returnn` we do (for constant inputs). And this is the right place.

Reply:
So the input to `Length` is the original input tensor which is not marked as `is_const`. This is why this is not done automatically.

Reply:
But the `zeros` produces a `Constant` which is always marked as `is_const` or not, even when the shape is not const? I think it should.
And the `fill_value` should also marked `is_const`, and then the `cast` as well, and then the `add`.

Reply:
The output of `zeros` is not marked as const.

```
>>> zeros_ = zeros(*size, dtype=dtype)
>>> naming.tensors[zeros_]
<TensorEntry name:? tensor:(Batch(3),) returnn_data:None>
>>> naming.tensors[zeros_].is_const
False
```

Reply:
This was done from inside `full`

Reply:
Also, `fill_value` is via `Length`, so not sure if it should be marked `is_const`.

Reply:
I removed `entry.is_const=False` as in your related comment and set `is_const=True` and `is_defined=True` in `SizeValue.as_tensor()`. Still, this does not work. If I set a breakpoint in `torch.full` when running `test_arange_from_lengths`, for both `zeros_` and `fill_value_` we have `is_const=False` and `is_defined=False`.

Reply:
> Also, `fill_value` is via `Length`, so not sure if it should be marked `is_const`.
> set `is_const=True` and `is_defined=True` in `SizeValue.as_tensor()`

Why `is_defined`? Not only that but also the buffer.

Then the `Length` output tensor should not be `is_const=False` anymore, right?

> Still, this does not work. If I set a breakpoint in `torch.full` when running `test_arange_from_lengths`, for both `zeros_` and `fill_value_` we have `is_const=False` and `is_defined=False`.

But where does this get set to `is_const=False`?

Reply:
Thanks a lot for your commits! I think they fixed this, so I'm resolving this thread.